modules/imgproc/src/hal_replacement.hpp

```diff
@@ -86,6 +86,40 @@ int my_hal_filterFree(cvhalFilter2D *context) {
  */
 struct cvhalFilter2D {};
 
+/**
+   @brief 2D filtering in a stateless manner
+   @param src_type source image type
```

Comment:
Nit, param descriptions has a different order compared how they listed in the argument list. It applies to all three added functions.